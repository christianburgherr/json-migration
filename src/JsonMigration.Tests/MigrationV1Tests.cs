using System.Text.Json.Nodes;
using FluentAssertions;
using Snapshooter.Xunit;

namespace JsonMigration.Tests;

public class MigrationV1Tests
{
    [Fact]
    public void Migrate_ShouldMigrateCorrect()
    {
        var migration = new MigrationV1();
        JsonObject rawJson = (JsonObject)JsonNode.Parse(File.ReadAllText("Resources/outdated.json"))!;

        var targetObject = new TestJsonObject();
        migration.Migrate(rawJson, targetObject);

        targetObject.Should().MatchSnapshot();
    }
}
