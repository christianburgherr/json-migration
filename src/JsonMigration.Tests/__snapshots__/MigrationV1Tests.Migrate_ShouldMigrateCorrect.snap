{
  "Version": 1,
  "NumberProperty": 11,
  "StringProperty": "AAA1",
  "ArrayProperty": [
    "A",
    "B",
    "C"
  ],
  "SubProperty": {
    "SubStringProperty": "AAA",
    "SubNumberProperty": 111
  }
}
